framework/src/modules/chain/chain.js

```diff
@@ -171,10 +171,29 @@ module.exports = class Chain {
 				this.options.genesisBlock,
 			);
 
+			this.channel.subscribe('app:state:updated', event => {
+				Object.assign(this.scope.applicationState, event.data);
+			});
+
+			this.logger.info('Modules ready and launched');
+			// After binding, it should immediately load blockchain
+			await this.processor.init(this.options.genesisBlock);
+
+			// Update Application State after processor is initialized
+			this.channel.invoke('app:updateApplicationState', {
+				height: this.blocks.lastBlock.height,
+				lastBlockId: this.blocks.lastBlock.id,
+				maxHeightPrevoted: this.blocks.lastBlock.maxHeightPrevoted || 0,
+				blockVersion: this.blocks.lastBlock.version,
+			});
+
 			// Deactivate broadcast and syncing during snapshotting process
-			if (this.options.loading.rebuildUpToRound) {
+			if (this.options.loading.rebuildUpToRound !== null) {
```

Comment:
does this need to be strictly `null`?

Reply:
Yes because of its default value. 

https://github.com/LiskHQ/lisk-sdk/blob/0ca424501ba2123815f37fe871d44748171e31b3/framework/src/modules/chain/defaults/config.js#L272